### ISSUE TITLE
how to put the data

### ISSUE THREAD

**Issue by aaahuaaa** [zihang-xu/ASC#2]
How can I encapsulate the data of atlases and feta2021 so that the program can successfully access them?

**Comment by zihang-xu** [comment]
You can download the dataset according to the link provided, and then prepare train.list and test.list files which record the file name. The training process can refer to the readme.md file, and more details can refer to the paper and supplementary materials.

**Comment by aaahuaaa** [comment]
Thank you ! However, I would like to ask how I can obtain the train.list and test.list files. It seems that these files are not provided in the original dataset. Do I need to generate them myself using a program?

**Comment by zihang-xu** [comment]
You are right! These two files need to be generated by yourself, and the basis of data splitting can refer to the paper.

**Comment by aaahuaaa** [comment]
Could you tell me what files are included in the path referred to by this "list". Does it only contain .nii files, or should .json files also be included in the list?

**Comment by zihang-xu** [comment]
It only contains .nii files.

**Comment by aaahuaaa** [comment]
In dataloader.py, when operating on the atlases dataset, it seems that directly converting case[:-8] to an int variable may cause problems. I would like to ask if the three datasets of atlases need to adhere to a specific naming format? (For example, one of my file naming conventions is: "/public/ASC-main/dataset/atlases/datalist/Chinese22_Study1/mask.nii.gz" , is it correct?)

**Comment by zihang-xu** [comment]
That was my negligence. According to your example, it should be "/public/ASC-main/dataset/atlases/label/Chinese22_Study1/mask.nii.gz" and "/public/ASC-main/dataset/atlases/image/Chinese22_Study1/parcellation.nii.gz)".
BTW, case[:-8] is used to extract the gestational age information from the file name.

**Comment by aaahuaaa** [comment]
Thank you very much for your timely and detailed response. However, I am still encountering an error when running dataloader.py with the statement **"self.age = int(case[:-8])"**. It seems that Python does not allow the use of the **int()** function to attempt to convert the file path"/public/ASC-main/dataset/atlases/datalist/Chinese22_Study1/mas"_(this path is case[:-8])_ into an integer.

**Comment by aaahuaaa** [comment]
BTW, I am wondering which one should I put in the image file: parcellation or srr?
![image](https://github.com/zihang-xu/ASC/assets/107868758/75ed7887-c78a-4d4b-bc52-6ec9d1c7d08c)
![image](https://github.com/zihang-xu/ASC/assets/107868758/be525894-c78e-4996-b5ec-0a639c16ad25)

**Comment by zihang-xu** [comment]
The list file was not saved, which caused me to remember something wrong. You can modify the code appropriately to match your list file, and I will upload the list file later. Atlas dataset should be a combination of three data sets, named in order, corresponding to "case" in the code, such as "srr01.nii.gz". "self.age" is actually the order of obtaining data files, because the gestational age and lesions of the three data sets are not directly reflected in the original file name, so I simply used the if-else structure to process them.
srr is image, parcellation is multi-class label, mask is outer contour (not the label)
If you have any questions, welcome.

**Comment by aaahuaaa** [comment]
I would like to ask about the feta2021 dataset, whether in the train.list and test.list, each line only needs to be written as "sub-XXX" (as shown in the picture) 

![image](https://github.com/zihang-xu/ASC/assets/107868758/097f49bc-9aad-4042-9183-24b2c1c0c2c3)

or if it also needs to be consistent with the atlases, where the label and image naming are different (one is parcellation and the other is srr).

**Comment by aaahuaaa** [comment]
I would like to ask about the feta2021 dataset, whether in the train.list and test.list, each line only needs to be written as "sub-XXX" (as shown in the picture) or if it also needs to be consistent with the atlases, where the label and image naming are different (one is parcellation and the other is srr).

**Comment by zihang-xu** [comment]
For feta2021 dataset, you can refer to this picture.
![e7431d2d822e78d463a949a2635221a](https://github.com/zihang-xu/ASC/assets/67997773/aaf6e5f5-e7f9-416d-a1d9-87fb95c34db0)

**Comment by aaahuaaa** [comment]
Based on the generated log.txt, it seems that my loss_inter is consistently showing as "nan". Below is my configuration. Can you advise me on which direction I should adjust?
(num_classes=8, max_epoch=100, batch_size=2, patch_size=[144, 144, 144], deterministic=1, base_lr=0.0001, epoch_gap=5, seed=1337, gpu='0', ema_decay=0.99, consistency_type='mse', consistency=200, consistency_rampup=100.0, zoom=1, crop=1)

**Comment by aaahuaaa** [comment]
It seems that when bs is less than 4, cut_mask will not be generated because int(batch_size_half/2) equals 0. However, my GPU memory is limited, and the program cannot run properly when bs=4. I would like to ask if there is a compromise solution (for example, can int(batch_size_half/2) be replaced with int(batch_size_half) here)?

in asc.py:
“cut_mask = mask_generator.generate_params(int(batch_size_half / 2), (volume_batch_t.shape[-3],volume_batch_t.shape[-2],volume_batch_t.shape[-1]))”

**Comment by zihang-xu** [comment]
The method of cutmix requires at least bs>=2 while considering the source and target domain requires bs>=4. Or you can remove the code of cutmix as described in the method section of the paper.

**Comment by aaahuaaa** [comment]
I would like to ask whether you used distributed multi-GPU training when implementing ASC. Since I only have two 3090 GPUs, I'm wondering if using distributed multi-GPU training would decrease performance.

**Comment by zihang-xu** [comment]
Our experience shows that distributed multi-GPU training with A100 will not reduce performance, and you can also try to verify it.

**Comment by aaahuaaa** [comment]
It seems that during testing, the Dice score for the first 15 samples and the last 25 samples are calculated separately. Do they correspond to 'abnormal' and 'normal' as mentioned in the paper.
Can I pick 15 samples with the **illness** list from your dataloader.py and put them into the first 15 of the test.list, and then pick 25 samples from the remaining normal codes and put them into the last 25 of the test.list?

**Comment by zihang-xu** [comment]
The code sequence corresponds to 15 normal and 25 abnormal cases, you can sample according to the paper. Besides, you can use **sampled_batch['pathological']** to determine whether it is abnormal.

**Comment by aaahuaaa** [comment]
It seems that ASC can only analyze and process three-dimensional data. 
Have you tried using ASC to process the CHAOS and MMWHS datasets?

**Comment by zihang-xu** [comment]
Unfortunately not. To test the project on two-dimensional data, you must modify the backbone, fft module, and cutmix module. Also, you can rewrite the code to fit your task.

**Comment by aaahuaaa** [comment]
Sorry to bother you again.
I've noticed that using SegResnet seems to occupy too much GPU memory. Is there a way to reduce memory usage without compromising performance (such as removing the reconstruction branch of SegResnet)? 
Additionally, are there any specific requirements for input images when using ASC, such as consistent size and spacing in terms of width, height, and depth?

**Comment by zihang-xu** [comment]
We haven't tried to modify the network architecture, maybe you could try distributed training?
The algorithm is designed without considering the effect of image size, and may have limitations when applied to different datasets. Welcome to discuss!
